### DISCUSSION TITLE
Get model by name

### DISCUSSION THREAD

**Discussion by nehalist** [prisma/prisma#19240] [Q&A]
Hey there, I've researched for a bit but couldn't find any answer to that: is it possible to get a model by its name? E.g.:

```js
const modelName = "user";
const model = prisma[modelName as keyof PrismaClient];
if (! model) {
  return;
}
// ...model.findUnique, etc
```

I couldn't figure out the type when using `find...`, since there doesn't seem to be a general `Delegate` type.

**Comment by ludralph** [comment]
Hi @nehalist 👋 

Can you please elaborate on your use case for why you need to get a model by its name?

You can access all models using `Prisma.dmmf.datamodel.models` but you should know that the `dmmf` is an internal api which is subject to change.

The output from `Prisma.dmmf.datamodel.models` is of the form:
```js
[
  {
    name: 'Pet',
    dbName: null,
    fields: [ [Object], [Object], [Object], [Object], [Object] ],
    primaryKey: null,
    uniqueFields: [],
    uniqueIndexes: [],
    isGenerated: false
  },
  {
    name: 'Owner',
    dbName: null,
    fields: [ [Object], [Object], [Object], [Object] ],
    primaryKey: null,
    uniqueFields: [],
    uniqueIndexes: [],
    isGenerated: false
  }
]
```
Based on the output you can get the model name.

**Reply by nurul3101** [reply]
@nehalist Did you had a chance to look at Raphael's response?

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
